### PR TITLE
docs: add section on error values to resource author's guide

### PR DESCRIPTION
**Pull request by BrianHicks**
Ping @ryane

### CODE REVIEW

File: docs_source/content/resource-authors-guide.md [hunk 1]
```diff
@@ -86,6 +86,14 @@ these rules:
 1. Otherwise, if there are any diffs which say that they contain a difference,
    the Status will always show as having changes.
 
+### Dealing with Errors
+
+The default `Status` implementation has a `SetError(error)` method. When called,
+it sets an appropriate error level and an internal error state that will be
```
Comment (ryane) [inline docs_source/content/resource-authors-guide.md]: should you also return the error if you call SetError?

File: docs_source/content/resource-authors-guide.md [hunk 2]
```diff
@@ -86,6 +86,14 @@ these rules:
...
+presented to the user. You can also use the common Go patterm of returning `nil,
+err` in your `Check` and `Apply` statements. Converge will call `SetError(err)`
+automatically in this case.
+
```
Comment (ryane) [inline docs_source/content/resource-authors-guide.md]: fyi, until #345 is merged, your error can be lost (regardless of whether you `SetError` or just return it) if the final plan phase `HasChanges`

File: docs_source/content/resource-authors-guide.md (hunk 1 above)
Comment (BrianHicks) [inline docs_source/content/resource-authors-guide.md]: good point, I've added a section to clarify in ebd589c

File: docs_source/content/resource-authors-guide.md (hunk 2 above)
Comment (BrianHicks) [inline docs_source/content/resource-authors-guide.md]: We'll consider that a blocker for this.

**Pull request merged**